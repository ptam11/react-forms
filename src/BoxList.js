import React, { Component } from 'react';
import Box from './Box';
import BoxForm from './BoxForm';
import uuid from 'uuid/v4';

class BoxList extends Component {
  constructor(props){
    super(props);
    this.state = {
      boxes:[{
        id: uuid(),
        width: 100,
        height: 100,
        backgroundColor: 'dodgerblue'
      },],
    }
    this.addBox = this.addBox.bind(this);
  }

  addBox(paramObj) {
    const newBox = {...paramObj, id: uuid()}
    this.setState(state => ({
      boxes: [...state.boxes, newBox]
    }));
  }

  render() {
    const theBoxes = this.state.boxes.map(box => (
      <Box key={box.id}
            height={box.height}
            width={box.width}
            backgroundColor={box.backgroundColor}
            />
            ))
            return (
              <div className="BoxList">
        {theBoxes}
        <BoxForm addBox={this.addBox}  />
      </div>
    );
  }

}

export default BoxList;
