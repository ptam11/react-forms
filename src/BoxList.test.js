import React from 'react';
import BoxList from './BoxList';
import { shallow, mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({
  adapter: new Adapter()
});


it('renders without errors', () => {
  shallow(<BoxList />);
});

it('snapshot matches', () => {
  let wrapper = shallow(<BoxList />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

xit('it renders correct message', () => {
  let wrapper = mount(<BoxList />);
  wrapper.setState({ msg: "Hello", color: "rebeccapurple" });
  let p_elem = wrapper.find('p').first();
  expect(p_elem.equals(<p>Hello</p>)).toEqual(true);
});

const shortList = [
  { msg: "Whiskey is a dog.", color: "green" },
  { msg: "Whiskey likes food.", color: "green" },
  { msg: "Whiskey likes sleeping.", color: "green" },
]

xit('it changes default message on click', () => {
  let wrapper = mount(<BoxList  answers={shortList}/>);
  wrapper.simulate("click");
  expect(wrapper.html()).toContain("Whiskey");
});