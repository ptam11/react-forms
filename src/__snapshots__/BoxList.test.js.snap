// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<div
  className="BoxList"
>
  <BoxForm
    addBox={[Function]}
  />
  <Box
    backgroundColor="dodgerblue"
    height={100}
    key="683c5abe-1b49-45ff-929c-4d32e6b57a45"
    width={100}
  />
</div>
`;
